### PR TITLE
Domains postbox renovation

### PR DESCRIPTION
**Pull request by JoosuaKoskinen**
#### What are the main changes in this PR?

- Adds support for 100% wide postboxes
- Makes the page look better
- Makes everything use AJAX
- Cleans up the codebase
- Improves error management
- Makes the page fully translatable
- Fixes time formats
- Makes all columns sortable
- Adds sniffing of customer managed dns
- Makes the changes made to the zone shown
- Adds ability to change the primary domain

This is not ready to merge yet. Note that this is just the domains postbox and the mails postbox PR will be separate. This PR will be updated and improved later.

### CODE REVIEW

**Comment by ottok** [comment]
Yay!

Will Pete test and review this?

**Comment by elguitar** [comment]
^ Changed the title, because @JoosuaKoskinen stated that this is not ready for merge.

**Comment by JoosuaKoskinen** [comment]
Renamed as this is ready for merge after @elguitar does his review. Also SWD still has some features coming related to this and I think we should wait for them before merging.

File: lib/domains-ajax.php [hunk 1]
```diff
@@ -42,6 +78,31 @@ function seravo_admin_change_zone_file() {
       return seravo_respond_error_json($response->get_error_message());
       wp_die();
     }
+
+    $response_decoded = json_decode($response, true);
+    if ( isset($response_decoded['diff']) && strlen($response_decoded['diff']) > 0 ) {
+      $records = Seravo_DNS_Table::fetch_dns_records('zone', $_REQUEST['domain']);
+      if ( ! isset($records['error']) ) {
+        $zone_diff = array();
+        $diff = explode("\n", $response_decoded['diff']);
```
Comment (elguitar) [inline lib/domains-ajax.php:87]: This whole diff-building could use a comment describing what it does in a whole.

File: lib/domain-tables.php [hunk 2]
```diff
@@ -0,0 +1,552 @@
...
+        $actions['view'] = sprintf($action_request, 'sniff', __('View', 'seravo'));
+        $actions['edit'] = sprintf($action_disabled, __('DNS not managed by Seravo.', 'seravo'), __('Edit', 'seravo'));
+      } else {
+        $actions['view'] = sprintf($action_disabled, __('This domain doesn\'t have a zone.', 'seravo'), __('View', 'seravo'));
+        $actions['edit'] = sprintf($action_disabled, __('This domain doesn\'t have a zone.', 'seravo'), __('Edit', 'seravo'));
+      }
+    }
+
+    if ( empty($item['primary']) ) {
+      $actions['primary'] = sprintf($action_request, 'primary', __('Make Primary', 'seravo'));
```
Comment (elguitar) [inline lib/domain-tables.php:79]: Maybe add `(experimental)` here? That way we can merge this even before the backend logic is clear.

We should then publish a new version as soon as possible. What do you think?

File: lib/domains-ajax.php (hunk 1 above)
Comment (JoosuaKoskinen) [reply lib/domains-ajax.php:89]: I added a few comments

File: lib/domain-tables.php (hunk 2 above)
Comment (JoosuaKoskinen) [reply lib/domain-tables.php]: `(experimental)` has been added

**Pull request merged**